crates/solver/src/driver/solver_settlements.rs

```diff
@@ -17,8 +38,11 @@ pub struct RatedSettlement {
     pub surplus: BigRational,     // In wei.
     pub earned_fees: BigRational, // In wei.
     pub solver_fees: BigRational, // In wei.
-    pub gas_estimate: U256,       // In gas units.
-    pub gas_price: BigRational,   // In wei per gas unit.
+    // protocol estimated gas usage
+    pub gas_estimate: U256, // In gas units.
+    // protocol estimated gas price
+    pub gas_price: BigRational, // In wei per gas unit.
+    pub gas_cost: GasCost,      // In wei.
```

Comment:
I don't think cost should be needed here. We have a gas estimate which can either bei SolverEstimated or ProtocolEstimated and together with the gas price this should allow computing the score (we could even enforce this gas estimate on the final transaction so it reverts if solvers lie about it, although this may not be fair given gas for pre/post interactions)